pkg/egresspolicy/manager.go

```diff
@@ -0,0 +1,461 @@
+		return
+	}
+
+	if podData, err = manager.getPodMetadata(pod, podIPs); err != nil {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.K8sPodName:   pod.Name,
+			logfields.K8sNamespace: pod.Namespace,
+		}).Error("Failed to get valid pod metadata, skipping update to Egress Policy")
+		return
+	}
+
+	if removeOld {
+		// Check if the pod was previously selected by any of the policies.
+		if policies, ok := manager.policyPods[podData.id]; ok {
+			for _, policy := range policies {
+				config := manager.policyConfigs[policy]
+				err := manager.deleteEgressMap(config, podData)
+				if err != nil {
+					log.WithError(err).Errorf("Error updating pod mapping for pod %v", podData.id)
+					return
+				}
+			}
+			delete(manager.policyPods, podData.id)
+			delete(manager.podDataStore, podData.id)
+		}
+	}
+
+	if upsertNew {
+		// Check if any of the current policies select this pod.
+		for _, config := range manager.policyConfigs {
+			if config.policyConfigSelectsPod(podData) {
+				err := manager.upsertPolicyPod(config, podData)
+				if err != nil {
+					log.WithError(err).Errorf("Error upserting pod mapping for pod %v", podData.id)
+					return
+				}
+			}
+		}
+	}
+}
+
+//OnUpdatePod is the event handler for pod updates.
+func (manager *Manager) OnUpdatePod(pod *slim_corev1.Pod, needsReassign bool) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	manager.OnUpdatePodLocked(pod, needsReassign, true)
+}
+
+//OnDeletePod is the event handler for pod deletions.
+func (manager *Manager) OnDeletePod(pod *slim_corev1.Pod) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	if len(manager.policyConfigs) == 0 {
+		return
+	}
+	id := types.NamespacedName{
+		Name:      pod.GetName(),
+		Namespace: pod.GetNamespace(),
+	}
+
+	podIPs := k8sUtils.ValidIPs(pod.Status)
+	if len(podIPs) == 0 {
+		return
+	}
+
+	var podData *podMetadata
+	var err error
+	if podData, err = manager.getPodMetadata(pod, podIPs); err != nil {
+		log.WithError(err).WithFields(logrus.Fields{
+			logfields.K8sPodName:   pod.Name,
+			logfields.K8sNamespace: pod.Namespace,
+		}).Error("failed to get valid pod metadata")
+		return
+	}
+
+	if policies, ok := manager.policyPods[id]; ok {
+		for _, policy := range policies {
+			config := manager.policyConfigs[policy]
+			err := manager.deleteEgressMap(config, podData)
+			if err != nil {
+				log.WithError(err).Errorf("Error deleting pod mapping for policy %v", config.id.Name)
+				return
+			}
+		}
+		delete(manager.policyPods, id)
+	}
+	delete(manager.podDataStore, id)
+}
+
+func (manager *Manager) getLocalPodsForNamespace(ns *slim_corev1.Namespace) []*slim_corev1.Pod {
+	var retPods []*slim_corev1.Pod
+
+	podStore := manager.storeGetter.GetStore("pod")
+	for _, podItem := range podStore.List() {
+		pod, ok := podItem.(*slim_corev1.Pod)
+		if !ok {
+			continue
+		}
+
+		if pod.Namespace != ns.Name {
+			continue
+		} else {
+			retPods = append(retPods, pod)
+		}
```

Comment:
No need for if-elses since the `continue` will skip execution
```suggestion
		if pod.Namespace != ns.Name {
			continue
		}
		retPods = append(retPods, pod)
```

Reply:
Fixed.